cmake/toolchain/raspberrypi.cmake

```diff
@@ -1,41 +1,31 @@
 ####
 # Raspberry PI Toolchain
 #
-# A toolchain file for the raspberrypi. This toolchain can be used to build against the raspberry pi embedded Linux
-# target. In order to use this toolchain, the raspberry pi build tools should be cloned onto a Linux build host. These
-# tools are available at: [https://github.com/raspberrypi/tools](https://github.com/raspberrypi/tools)
-#
-# Typically these tools are cloned to `/opt/rpi/`.  If they are cloned elsewhere, the user must set the environment
-# variable `RPI_TOOLCHAIN_DIR` to the full path to the `arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf` directory before
-# running CMake or `fprime-util generate`.
-#
-# e.g. should the user install the tools in ``/home/user1` then the environment variable might be set using
-# `export RPI_TOOLCHAIN_DIR=/home/user/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf/`
+# This is a toolchain for the Raspberry Pi. This toolchain can be used ot build
+# against the Raspberry Pi embedded Linux target. In order to use this toolchain,
+# the Raspbery Pi cross-compiler should be installed on a Linux host. These
+# tools are installable as follows:
+#   sudo apt install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf  gdb-multiarch
 ####
-# Set system name
-set(CMAKE_SYSTEM_NAME "Linux")
-set(CMAKE_SYSTEM_PROCESSOR "arm")
 
-# Location of pi toolchain
-set(RPI_TOOLCHAIN "$ENV{RPI_TOOLCHAIN_DIR}")
-if ("${RPI_TOOLCHAIN}" STREQUAL "")
-    set(RPI_TOOLCHAIN "/opt/rpi/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf")
-endif()
-# Check toolchain directory exists
-IF(NOT EXISTS "${RPI_TOOLCHAIN}")
-    message(FATAL_ERROR "RPI toolchain not found at ${RPI_TOOLCHAIN}. Install it, set RPI_TOOLCHAIN_DIR environment variable, or choose another toolchain")
-endif()
-message(STATUS "Using RPI toolchain at: ${RPI_TOOLCHAIN}")
-# specify the cross compiler
-set(CMAKE_C_COMPILER "${RPI_TOOLCHAIN}/bin/arm-linux-gnueabihf-gcc")
-set(CMAKE_CXX_COMPILER "${RPI_TOOLCHAIN}/bin/arm-linux-gnueabihf-g++")
+# Set the system information
+set(CMAKE_SYSTEM_NAME       Linux)
+set(CMAKE_SYSTEM_PROCESSOR  arm)
+set(CMAKE_SYSTEM_VERSION    1)
 
-# where is the target environment
-set(CMAKE_FIND_ROOT_PATH  "${RPI_TOOLCHAIN}")
+# Set the GNU ARM toolchain
+set(CMAKE_AR            arm-linux-gnueabihf-ar)
+find_program(CMAKE_C_COMPILER NAMES arm-linux-gnueabihf-gcc PATHS ENV RPI_TOOLCHAIN_DIR PATH_SUFFIXES bin REQUIRED)
+find_program(CMAKE_CXX_COMPILER NAMES arm-linux-gnueabihf-g++ PATHS ENV RPI_TOOLCHAIN_DIR PATH_SUFFIXES bin REQUIRED)
+message(STATUS "[arm-linux] C Compiler: ${CMAKE_C_COMPILER}")
+message(STATUS "[arm-linux] CXX Compiler: ${CMAKE_CXX_COMPILER}")
+set(CMAKE_ASM_COMPILER  arm-linux-gnueabihf-as)
```

Comment:
Don't we need to run a `find_program` for every tool listed here, not just `g++` and `gcc`?

Reply:
Also, there is one true spelling error. The rest should be added to the expect list.

Reply:
What other tools would we want to run a `find_program` for?

Reply:
How do I resolve the spell check errors?

Reply:
For example each line should be a find_program:
```
set(CMAKE_ASM_COMPILER  arm-linux-gnueabihf-as)
```
->
```
find_program(CMAKE_ASM_COMPILER NAMES arm-linux-gnueabihf-as PATHS ENV RPI_TOOLCHAIN_DIR PATH_SUFFIXES bin REQUIRED)
```

The reason is that you'd likely want to ensure toy are pulling a consistent set of tools from the same directory.

To fix spelling errors:

```
db
multiarch
objcopy
objdump
```

Can be added to `expect.txt` under `.github/actions/spelling`

```
Raspbery 
```

Appears to be a true misspelling.

Reply:
Sorry for the delay, been out traveling these past few weeks! The spell check errors should be resolved now. Also replaced `set` with `find_program` on the lines where it looks for the ARM toolchain